drivers/i2c/meson/driver.h

```diff
@@ -17,10 +17,10 @@
 #include "gpio.h"
 #include "clk.h"
 
-enum data_direction {
-    DATA_DIRECTION_WRITE = 0x0,
-    DATA_DIRECTION_READ = 0x1
-};
+typedef uint8_t data_direction_t;
+#define DATA_DIRECTION_WRITE    (0x0)
+#define DATA_DIRECTION_READ     (0x1)
+#define DATA_DIRECTION_READC    (0x3)   // Allows bitwise check for read or readc
```

Comment:
Is `DATA_DIRECTION_READC` ever used? It's used to set `data_direction` but then `data_direction` is only ever checked if the first bit is set so it seems like this define isn't needed?

Reply:
It's checked implicitly. Whenever an operation doesn't need one or the other we check if the DATA_DIRECTION & 1 is set, and the one place this directly matters is on line 448 in i2c.c

```
            if (i2c_ifState.rw_remaining == 1 && i2c_ifState.data_direction == DATA_DIRECTION_READ) {
                // Write data end on last byte of a read.
                meson_token = MESON_I2C_TOKEN_DATA_END;
```

If READC is set, the DATA_END token is not transmitted. The only purposes of READC is to be distinct from READ on this line.

Reply:
(for meson, anyway. READC is more important for Cheshire for example)

Reply:
yea but you never differentiate between `DATA_DIRECTION_READ` and `DATA_DIRECTION_READC` and so I don't see the purpose of having two defines.

Reply:
oh nevermind, ignore me sorry.